deep_qa/common/util.py

```diff
@@ -41,16 +41,21 @@ def pad_sequence_to_length(sequence: List,
     padding_on_right : bool, default=True
         When we add padding tokens (or truncate the sequence), should we do it on the right or
         the left?
+
+    Returns
+    -------
+    padded_sequence : List
     """
     if padding_on_right:
-        del sequence[desired_length:]
+        padded_sequence = sequence[:desired_length]
```

Comment:
I think it's cleaner to not modify the input list - that way we don't get subtle bugs like we had with the decomposable attention padding issue that you ran into.  I switch this to return a new list instead of modifying the input list.